src/lib/notcurses.c

```diff
@@ -532,6 +532,30 @@ make_ncpile(notcurses* nc, ncplane* n){
   return ret;
 }
 
+
+static inline size_t  ncplane_sizeof_cellarray( unsigned rows, unsigned cols)
+{
+  size_t fbsize;
+  size_t size;
+
+  // (nat) This protects against size_t overflow and also checks
+  // that dimensions are not zero en-passant. Why ?
+  // Assume: size_t is 16 bit, unsigned is 8 bit (UINT_MAX: 0xFF)
+  // nccell is sizeof( *p->fb): 0x10 (128 bit)
+  //
+  // 0xFF * 0xFF * 0x10 = 0xFE010, but overflows so: 0xE010
+  // 0x3F * 0x3F * 0x10 = 0xf810 is the biggest square possible
+
+  size = (size_t) cols * (size_t) rows;
```

Comment:
i feel that

`if((uintmax_t)cols * rows > SIZE_MAX){`

is more obvious and more robust, if we're going to go down this path.

Reply:
Feelings can be treacherous. if `uintmax_t` is the same size as `size_t` then the multiplication is still in overflow territory. The comparison `> SIZE_MAX` would have to be `SIZE_MAX  / sizeof( struct nccell)`, otherwise you'd overflow later on.

Reply:
doh, of course